### ISSUE TITLE
Update tractability facet data returned by API

### ISSUE THREAD

**Issue by andrewhercules** [opentargets/platform#1751]
With the new `target` ETL pipeline and API response, we now have updated tractability facets available on the disease associations page.

```
query targetsAssociatedWithADiseaseFacets{
  disease(efoId:"EFO_0000676"){
    id
    name
    associatedTargets{
      count
      aggregations{
        aggs {
          uniques
          name
          rows{
            key
            uniques
          }
        }
      }
    }
  }
}
```
Sample query: [EFO_0000676](https://api-beta-dot-open-targets-eu-dev.ew.r.appspot.com/api/v4/graphql/browser?query=query%20targetsAssociatedWithADiseaseFacets%7B%0A%20%20disease%28efoId%3A%22EFO_0000676%22%29%7B%0A%20%20%20%20id%0A%20%20%20%20name%0A%20%20%20%20associatedTargets%7B%0A%20%20%20%20%20%20count%0A%20%20%20%20%20%20aggregations%7B%0A%20%20%20%20%20%20%20%20aggs%20%7B%0A%20%20%20%20%20%20%20%20%20%20uniques%0A%20%20%20%20%20%20%20%20%20%20name%0A%20%20%20%20%20%20%20%20%20%20rows%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20key%0A%20%20%20%20%20%20%20%20%20%20%20%20uniques%0A%20%20%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%0A%7D)

In preparation for the 21.09 release, can we please make two adjustments to the tractability facet data returned by the GraphQL API?

- [ ] Add a facet, `tractabilityOtherModalities`, that includes the three other clinical modalities buckets — note: the buckets are prefixed with the `OC` modality

- [ ] Adjust the API response so that the order of the tractability buckets match the input file — this is similar to the work done in #1672

**Comment by JarrodBaker** [comment]
Hi Andrew

We don't have the capacity to include this in the 21.09 release.

**Comment by andrewhercules** [comment]
Sure, no worries @JarrodBaker. I have re-tagged for the 21.11 release 😁

**Comment by JarrodBaker** [comment]
Can you please provide an example you mean by `the order of the tractability buckets match the input file` @andrewhercules . What output would you expect to see in the API response?

**Comment by andrewhercules** [comment]
@JarrodBaker, if possible can we please return the tractability assessments using the same order that we return when querying the `target` endpoint (e.g. [ESR1 query](https://api.platform.opentargets.org/api/v4/graphql/browser?query=query%20targetTractabilityInformation%20{%20target(ensemblId:%20%22ENSG00000091831%22)%20{%20id%20approvedSymbol%20tractability{%20modality%20id%20value%20}%20}%20})).

So for example:

```
"tractability": [
        {
          "modality": "SM",
          "id": "Approved Drug"
        },
        {
          "modality": "SM",
          "id": "Advanced Clinical"
        },
        {
          "modality": "SM",
          "id": "Phase 1 Clinical"
        },
        {
          "modality": "SM",
          "id": "Structure with Ligand"
        },
        {
          "modality": "SM",
          "id": "High-Quality Ligand"
        },
        {
          "modality": "SM",
          "id": "High-Quality Pocket"
        },
        {
          "modality": "SM",
          "id": "Med-Quality Pocket"
        },
        {
          "modality": "SM",
          "id": "Druggable Family"
        },
        {
          "modality": "AB",
          "id": "Approved Drug"
        },
        {
          "modality": "AB",
          "id": "Advanced Clinical"
        },
        {
          "modality": "AB",
          "id": "Phase 1 Clinical"
        },
        {
          "modality": "AB",
          "id": "UniProt loc high conf"
        },
        {
          "modality": "AB",
          "id": "GO CC high conf"
        },
        {
          "modality": "AB",
          "id": "UniProt loc med conf"
        },
        {
          "modality": "AB",
          "id": "UniProt SigP or TMHMM"
        },
        {
          "modality": "AB",
          "id": "GO CC med conf"
        },
        {
          "modality": "AB",
          "id": "Human Protein Atlas loc"
        },
        {
          "modality": "PR",
          "id": "Approved Drug"
        },
        {
          "modality": "PR",
          "id": "Advanced Clinical"
        },
        {
          "modality": "PR",
          "id": "Phase 1 Clinical"
        },
        {
          "modality": "PR",
          "id": "Literature"
        },
        {
          "modality": "PR",
          "id": "UniProt Ubiquitination"
        },
        {
          "modality": "PR",
          "id": "Database Ubiquitination"
        },
        {
          "modality": "PR",
          "id": "Half-life Data"
        },
        {
          "modality": "PR",
          "id": "Small Molecule Binder"
        },
        {
          "modality": "OC",
          "id": "Approved Drug"
        },
        {
          "modality": "OC",
          "id": "Advanced Clinical"
        },
        {
          "modality": "OC",
          "id": "Phase 1 Clinical"
        }
      ]
```

The most important thing is that the clinical assessments where `id` is one of `Approved Drug`, `Advanced Clinical` or `Phase 1 Clinical` appear at the top of the list of facet options for each modality.

**Comment by JarrodBaker** [comment]
Let me see if I'm understanding you correctly @andrewhercules . Given I'm looking at a disease, say MONDO_0001023, when I look at the associations page and select the facet I currently see:

![image.png](https://images.zenhubusercontent.com/5ec7caaa0e8174edc9bdf7a6/a195d671-9050-4689-a931-41018ab97066)

We want to have this list ordered such that the first three entries are `Approved Drug`, `Advanced Clinical`, `Phase 1 Clinical`, then the rest depending on what is available?

We can do custom orderings, but these endpoints are already _sloooow_. We'll be adding additional computing overhead and thereby slowing them down further. Is this something we want to pursue now or wait until we have better performance on the queries?

**Comment by andrewhercules** [comment]
Hi @JarrodBaker!

Not to worry, the ordering of the tractability buckets can wait until we have made the queries more performant.

Is it possible to just expose the "Other Clinical" buckets starting with "OC"?

**Comment by JarrodBaker** [comment]
Just testing the new functionality now and we should have the new facets exposed.

**Comment by andrewhercules** [comment]
Ticket closed as work completed and released in `21.11`

Thank you @JarrodBaker! 😄
